### PR TITLE
(update) bump numpy version for python 3.12 compatibility

### PR DESCRIPTION
**Pull request by amontagne**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ aenum
 pydantic
 pyyaml==6.0.1
 sympy==1.12
-numpy==1.23.*
+numpy==1.26.4
```
Comment (m-prl) [inline requirements.txt:5]: appabuild needs a 1.23.* numpy version, and requires apparun. Can we safely update numpy on both repo and keep a compatiblity with python 3.9 ?

File: .gitignore [hunk 2]
```diff
@@ -19,5 +19,9 @@ venv.bak/
 *.idea
 *.iml
 
-# Outputs
-outputs/
\ No newline at end of file
+# Data
+data/
```
Comment (m-prl) [inline .gitignore:23]: will conflict with last PR, no?

Comment (amontagne) [reply .gitignore:23]: i will remove this change

File: requirements.txt (hunk 1 above)
Comment (m-prl) [reply requirements.txt:5]: Python>=3.10 required, seems acceptable but documentation needs to be updated

**Pull request merged**